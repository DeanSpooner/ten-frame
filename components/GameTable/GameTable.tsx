import React, { useEffect, useState } from "react";
import { View } from "react-native";
import gameTableStyles from "./GameTable.styles";

interface GameTableProps {
  counter: number;
}

const GameTable: React.FC<GameTableProps> = ({ counter }) => {
  const [randomColour, setRandomColour] = useState<string>("red");
  const lengthOfFrame: number = 10;
  const arrayOfCounters: number[] = Array.from(Array(lengthOfFrame)).map(
    (e, i) => i + 1
  );

  useEffect(() => {}, []);

  useEffect(() => {
    const generatedColour: string = Math.floor(Math.random() * 16777215)
      .toString(16)
      .padStart(6, "0");
    setRandomColour(`#${generatedColour}`);
  }, [counter]);

  return (
    <View style={gameTableStyles.container}>
      {arrayOfCounters.map((num) => (
        <View style={gameTableStyles.cell} key={num}>
          <View
            //@ts-ignore
            style={
              counter < num
                ? ""
                : [gameTableStyles.counter, { backgroundColor: randomColour }]
            }
          ></View>
        </View>
      ))}
    </View>
  );
};

export default GameTable;
