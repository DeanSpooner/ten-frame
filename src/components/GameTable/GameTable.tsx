import React, { useEffect, useState } from "react";
import { View } from "react-native";
import { randomColourGenerator } from "../../helpers/randomColourGenerator";
import { arrayMaker } from "../../helpers/arrayMaker";
import gameTableStyles from "./GameTable.styles";

interface GameTableProps {
  counter: number;
}

const GameTable: React.FC<GameTableProps> = ({ counter }) => {
  const [randomColour, setRandomColour] = useState<string>("red");
  const lengthOfFrame: number = 10;

  useEffect(() => {
    setRandomColour(`#${randomColourGenerator()}`);
  }, [counter]);

  return (
    <View style={gameTableStyles.container}>
      {arrayMaker(lengthOfFrame).map((num) => (
        <View style={gameTableStyles.cell} key={num}>
          <View
            //@ts-ignore
            style={
              counter < num
                ? ""
                : [gameTableStyles.counter, { backgroundColor: randomColour }]
            }
          ></View>
        </View>
      ))}
    </View>
  );
};

export default GameTable;
