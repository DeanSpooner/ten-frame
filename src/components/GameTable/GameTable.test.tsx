import React from "react";
import renderer from "react-test-renderer";
import GameTable from "./GameTable";

describe("<GameTable />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<GameTable counter={0} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("has 10 children", () => {
    const tree = renderer.create(<GameTable counter={0} />).toJSON();
    expect(tree.children.length).toBe(10);
  });
});
