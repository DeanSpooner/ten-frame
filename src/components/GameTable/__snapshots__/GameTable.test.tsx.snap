// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameTable /> renders correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "center",
      "maxWidth": 360,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#000",
        "borderWidth": 3,
        "height": 72,
        "justifyContent": "center",
        "width": 72,
      }
    }
  >
    <View
      style=""
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#000",
        "borderWidth": 3,
        "height": 72,
        "justifyContent": "center",
        "width": 72,
      }
    }
  >
    <View
      style=""
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#000",
        "borderWidth": 3,
        "height": 72,
        "justifyContent": "center",
        "width": 72,
      }
    }
  >
    <View
      style=""
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#000",
        "borderWidth": 3,
        "height": 72,
        "justifyContent": "center",
        "width": 72,
      }
    }
  >
    <View
      style=""
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#000",
        "borderWidth": 3,
        "height": 72,
        "justifyContent": "center",
        "width": 72,
      }
    }
  >
    <View
      style=""
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#000",
        "borderWidth": 3,
        "height": 72,
        "justifyContent": "center",
        "width": 72,
      }
    }
  >
    <View
      style=""
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#000",
        "borderWidth": 3,
        "height": 72,
        "justifyContent": "center",
        "width": 72,
      }
    }
  >
    <View
      style=""
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#000",
        "borderWidth": 3,
        "height": 72,
        "justifyContent": "center",
        "width": 72,
      }
    }
  >
    <View
      style=""
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#000",
        "borderWidth": 3,
        "height": 72,
        "justifyContent": "center",
        "width": 72,
      }
    }
  >
    <View
      style=""
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#000",
        "borderWidth": 3,
        "height": 72,
        "justifyContent": "center",
        "width": 72,
      }
    }
  >
    <View
      style=""
    />
  </View>
</View>
`;
